### PR TITLE
Moved to system timekeeping

### PR DESCRIPTION
**Pull request by jamesrisberg**
Transferred timekeeping to the system clock and unified the audio/video code paths for processing frames

### CODE REVIEW

**Comment by brion** [comment]
reverted -- broke seeking in webm files

File: Classes/OGVPlayerState.m [hunk 1]
```diff
@@ -154,12 +156,14 @@ -(BOOL)paused
 
 -(float)playbackPosition
 {
-    // @todo use alternate clock provider for video-only files
+    double position = 0.0;
     if (playing) {
-        return audioFeeder.playbackPosition + initialAudioTimestamp;
+        position = CACurrentMediaTime() - initTime + offsetTime;
```
Comment (brion) [inline Classes/OGVPlayerState.m]: Using a clock unrelated to the audio clock means audio is out of sync. Need to always use the audio clock if there is audio.

File: Classes/OGVPlayerState.m [hunk 2]
```diff
@@ -217,12 +225,8 @@ - (void)initPlaybackState
     assert(decoder.dataReady);
     
     frameEndTimestamp = 0.0f;
-    
-    dispatch_async(drawingQueue, ^() {
-        if ([delegate respondsToSelector:@selector(ogvPlayerStateDidPlay:)]) {
-            [delegate ogvPlayerStateDidPlay:self];
-        }
-    });
```
Comment (brion) [inline Classes/OGVPlayerState.m]: We've lost a delegate method call-through.

File: Classes/OGVPlayerState.m [hunk 3]
```diff
@@ -111,33 +120,26 @@ -(void)seek:(float)time
     }
     dispatch_async(decodeQueue, ^() {
         if (decoder && decoder.seekable) {
-            BOOL wasPlaying = !self.paused;
-            if (wasPlaying) {
+            wasPlayingBeforeSeek = !self.paused;
```
Comment (brion) [inline Classes/OGVPlayerState.m]: This seems to get confused when running multiple seeks, such as scrubbing in the example app. In the current code, seeking to a keypoint is a single atomic operation on the decodeQueue thread, but in this code, the seeking operation extends beyond that to further call-ins as we iterate through audio and video packets looking for our final target. As a result, when I scrub, the second seek sees that we're paused and sets wasPlayingBeforeSeek to false, so there's no continuation after seeking.

File: Classes/OGVPlayerState.m (hunk 2 above)
Comment (brion) [inline Classes/OGVPlayerState.m]: hmm, ok that seems to have been a dupe? meh :D

**Comment by brion** [comment]
Reworking this on 'timing' branch; some improvements but still some regressions seeking on audio-only files.

**Comment by brion** [comment]
Ok with this code plus additional fixes I think we now have correctly working play & seek with audio-only, video-only, and audio+video :D Thanks @jamesrisberg for the contrib patches and for helping guinea-pig test the library ;)

**Comment by jamesrisberg** [comment]
Woo! Sorry about the incomplete patch, but glad it's all up to snuff now! Always happy to contribute 😬

**Comment by brion** [comment]
No worries. Showed several places where my code wasn't up to snuff to begin with and I had to fix em... ;)

**Pull request merged**